### ISSUE TITLE
5.1.0: pytest fails in 4 units

### ISSUE THREAD

**Issue by kloczek** [celery/vine#106]
I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix> using `installer` module
- run pytest with $PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- build is performed in env which is *`cut off from access to the public network`* (pytest is executed with `-m "not network"`)

<details>
<summary>Here is pytest output:</summary>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-vine-5.1.0-4.fc36.x86_64/usr/lib64/python3.9/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-vine-5.1.0-4.fc36.x86_64/usr/lib/python3.9/site-packages
+ /usr/bin/pytest -ra -m 'not network'
============================= test session starts ==============================
platform linux -- Python 3.9.18, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/tkloczko/rpmbuild/BUILD/vine-5.1.0
configfile: setup.cfg
testpaths: t/unit/
collected 42 items

t/unit/test_abstract.py ..                                               [  4%]
t/unit/test_funtools.py .........                                        [ 26%]
t/unit/test_promises.py ...........................                      [ 90%]
t/unit/test_synchronization.py FFFF                                      [100%]

=================================== FAILURES ===================================
__________________________ test_barrier.test_evaluate __________________________

self = <t.unit.test_synchronization.test_barrier object at 0x7f5a978c32b0>

    def test_evaluate(self):
>       x = barrier(self.ps)
E       AttributeError: 'test_barrier' object has no attribute 'ps'

t/unit/test_synchronization.py:16: AttributeError
__________________________ test_barrier.test_reverse ___________________________

self = <t.unit.test_synchronization.test_barrier object at 0x7f5a978c3fd0>

    def test_reverse(self):
        callback = Mock()
>       x = barrier(self.ps, callback=promise(callback))
E       AttributeError: 'test_barrier' object has no attribute 'ps'

t/unit/test_synchronization.py:34: AttributeError
___________________________ test_barrier.test_cancel ___________________________

self = <t.unit.test_synchronization.test_barrier object at 0x7f5a978c3b20>

    def test_cancel(self):
>       x = barrier(self.ps)
E       AttributeError: 'test_barrier' object has no attribute 'ps'

t/unit/test_synchronization.py:41: AttributeError
___________________________ test_barrier.test_throw ____________________________

self = <t.unit.test_synchronization.test_barrier object at 0x7f5a978c39d0>

    def test_throw(self):
>       x = barrier(self.ps)
E       AttributeError: 'test_barrier' object has no attribute 'ps'

t/unit/test_synchronization.py:50: AttributeError
=========================== short test summary info ============================
FAILED t/unit/test_synchronization.py::test_barrier::test_evaluate - Attribut...
FAILED t/unit/test_synchronization.py::test_barrier::test_reverse - Attribute...
FAILED t/unit/test_synchronization.py::test_barrier::test_cancel - AttributeE...
FAILED t/unit/test_synchronization.py::test_barrier::test_throw - AttributeEr...
========================= 4 failed, 38 passed in 0.16s =========================
```
</details>

<details>
<summary>List of installed modules in build env:</summary>

```console
Package            Version
------------------ -----------
build              1.1.1
distro             1.9.0
exceptiongroup     1.1.3
importlib_metadata 7.0.1
iniconfig          2.0.0
installer          0.7.0
packaging          24.0
pluggy             1.4.0
pyproject_hooks    1.0.0
pytest             8.1.1
python-dateutil    2.9.0.post0
setuptools         69.1.1
tokenize_rt        5.2.0
tomli              2.0.1
wheel              0.43.0
zipp               3.17.0
```
</details>

Please let me know if you need more details or want me to perform some diagnostics.

**Comment by Nusnus** [comment]
Hey there, thank you for reporting this issue.
May I ask for a contribution to fix this?

I’d happily review a PR to help get this done quickly.

Thank you.

**Comment by mgorny** [comment]
This is already fixed in cf9b3979173ff22a4a410c4da6cfdad878eced8c.
